core-bundle/contao/controllers/BackendMain.php

```diff
@@ -238,22 +238,25 @@ protected function compileTemplateData(array $data): array
 		}
 
 		$container = System::getContainer();
+		$request = $container->get('request_stack')->getCurrentRequest();
+		$renderMainOnly  = $request->query->has('popup') || $request->headers->get('turbo-frame') === 'contao-main';
 
 		$data['theme'] = Backend::getTheme();
 		$data['language'] = $GLOBALS['TL_LANGUAGE'];
 		$data['title'] = StringUtil::specialchars(preg_replace('/^\s*›\s*|\s*›\s*$/u', '', strip_tags(preg_replace('/<span.*?>/', ' › ', $data['title'] ?? ''))));
 		$data['host'] = Backend::getDecodedHostname();
 		$data['charset'] = System::getContainer()->getParameter('kernel.charset');
 		$data['home'] = $GLOBALS['TL_LANG']['MSC']['home'];
-		$data['isPopup'] = Input::get('popup');
+		$data['isPopup'] = $request->query->has('popup');
```

Comment:
```suggestion
		$data['isPopup'] = $request->query->has('popup') && $request->query->get('popup');
```

That would be the exact equivalent of `Input::get('popup')`. Because the query parameter could theoretically be `&popup=` or `&popup=0`.

Reply:
We can just use `get()` then.